src/notification.rs

```diff
@@ -38,70 +41,66 @@ pub struct MustacheData {
     sites: Vec<Url>,
 }
 
-impl MustacheData {
-    pub fn new(commit: Option<String>, changed_urls: Vec<Url>) -> Self {
-        let mut sites = changed_urls;
-        sites.sort_unstable();
-        sites.dedup();
-
-        let mut hosts = sites
-            .iter()
-            .filter_map(Url::host_str)
-            .map(std::string::ToString::to_string)
-            .collect::<Vec<_>>();
-        hosts.dedup();
-
-        let singlehost = if let [single] = hosts.as_slice() {
+impl From<Summary> for MustacheData {
+    fn from(summary: Summary) -> Self {
+        let changed_hosts = summary.changed_hosts.into_keys().collect::<Vec<_>>();
```

Comment:
looks a bit ... ugh. Just thinking out load: 
```suggestion
        let changed_hosts = summary.changed_hosts.hostnames_as_vec();
```

Im writing because I am not sure what exactly is happening here which is not a good sign. Granted, I'm not proficient in rust yet but: This line expects you to know that `changed_hosts` is a `BTreeMap`. `into_keys()` hints that the values are dismissed? What is key and what is values? Are we dismissing the host names and keep the changes?

Reply:
`MustacheData` is legacy notification related stuff that is deprecated. This change ensures it's still working before it is removed.

https://github.com/EdJoPaTo/website-stalker/blob/9cdca0efa1206fec559fd3e8b2cccaca6a41e68f/src/summary.rs#L17

`changed_hosts` contains the hosts as keys with the URLs on the given host key as value. This is new, the notification features only had the hosts. `HashMap` also has `into_keys`. As the value is not used after this moving it is better performance wise. Therefore, `into_keys()` is used instead of `keys()`.